src/cmd/ls.rs

```diff
@@ -373,6 +373,10 @@ pub fn run_cmd<W: Write>(mut w: &mut W, cwd: &Path, cmd: &Cmd, project: &Project
                 if !ui::show_artifact(&name, art, &pat_case, &cmd.search_settings) {
                     continue;
                 }
+                if !match_made && !fmt_set.long && cmd.ty == OutType::List {
```

Comment:
That's what I tried originally, but it doesn't work. I'm not entirely clear on where, but somewhere `names` is being populated prior to that if statement. I used a `println!` just prior to the if statement to print `names`. When no options are set, `names` contains only the input search query. When `-p` is set, names is populated with what I believe is the full list of artifacts. 
![artifact output ls bug](https://cloud.githubusercontent.com/assets/14239437/23232193/5b159e88-f918-11e6-9c89-708475125ddd.PNG)